src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,513 @@
+            time = taskLine.substring(index + 3);
+        }
+
+        switch (taskType) {
+        case "D":
+            task = new Deadline(description, time);
+            if (taskDone.equals("√")) {
+                task.markAsDone();
+            }
+        case "E":
+            task = new Event(description, time);
+            if (taskDone.equals("√")) {
+                task.markAsDone();
+            }
+        default:
+            task = new ToDo(description);
+            if (taskDone.equals("√")) {
+                task.markAsDone();
+            }
+        }
+        return task;
+    }
+
+    public static String convertTodotoLine(ToDo task) {
+        if (task.isDone()) {
+            return ("T | √ | " + task.getDescription());
+        } else {
+            return ("T | X | " + task.getDescription());
+        }
+    }
+
+    public static String convertDeadlinetoLine(Deadline task) {
+        if (task.isDone()) {
+            return ("D | √ | " + task.getDescription() + " | " + task.getBy());
+        } else {
+            return ("D | X | " + task.getDescription() + " | " + task.getBy());
+        }
+    }
+
+    public static String convertEventtoLine(Event task) {
+        if (task.isDone()) {
+            return ("E | √ | " + task.getDescription() + " | " + task.getAt());
+        } else {
+            return ("E | X | " + task.getDescription() + " | " + task.getAt());
+        }
+    }
+
+    public static void initializeList(ArrayList<Task> tasks) throws FileNotFoundException {
+        File f = new File(LIST_STORAGE_FILE);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            tasks.add(convertLinetoTask(s.nextLine()));
+        }
+        printList(tasks, tasks.size());
+    }
+
+    public static void initializeTaskLineList(ArrayList<String> taskLines) throws FileNotFoundException {
+        File f = new File(LIST_STORAGE_FILE);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            taskLines.add(s.nextLine());
+        }
+    }
+
+    public static void appendToFile(String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(LIST_STORAGE_FILE, true);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    public static void clearFile() throws IOException {
+        FileWriter fw = new FileWriter(LIST_STORAGE_FILE);
+        fw.write("");
+        fw.close();
+    }
+
+    public static void storetoHardDisk(ArrayList<String> taskLines) {
+        try {
+            clearFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+        for (String taskLine : taskLines) {
+            try {
+                appendToFile(taskLine + System.lineSeparator());
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+    }
+
+    public static void main(String[] args){
```

Comment:
Avoid methods with longer than 30 lines. This method could perhaps be modularised and broken down into simpler constructs.